### ISSUE TITLE
Datetimepicker a11y - trigger with keyboard

### ISSUE THREAD

**Issue by jq-87** [buefy/buefy#2879]
### Overview of the problem

**Buefy** version: 0.9.3
**Vuejs** version: 2.6.11
**OS/Browser**: Chrome

### Steps to reproduce
1. Navigate form elements with keyboard tab key
2. When you focus the date field - the date ui does not come up. Click or touch is required.

### Expected behavior

Users need to access all controls with keyboard.

### Actual behavior
It is currently not possible to trigger date ui when navigating with keyboard.
https://buefy.org/documentation/datetimepicker/

**Comment by service-paradis** [comment]
I dont think the dropdown should expand only by focusing the element.
If you press the `enter` key while the input have focus, it will display the dropdown. Looks to be the expected behavior.

If you take a look at https://www.w3.org/TR/wai-aria-practices-1.1/examples/dialog-modal/datepicker-dialog.html, the dialog will open only when pressing enter (or space) while focusing on the date button. The dialog will not open only by focusing on it.

**Comment by jq-87** [comment]
FYI space bar does not work either. I can't seem to enable the picker at all with keyboard only.

<img width="996" alt="Screen Shot 2021-02-25 at 4 09 12 PM" src="https://user-images.githubusercontent.com/609653/109217664-18c6c400-7784-11eb-9fed-a79a0a5c6989.png">
